Enumerable.rb

```diff
@@ -0,0 +1,165 @@
+module Enumerable
+    def my_each
+        return to_enum unless block_given?
+        n= self.length
+        x = 0
+        while x < n
+            yield(self[x])
+            x += 1
+        end
+         self
+    end
+    def my_each_with_index
+        return to_enum unless block_given?
+        n= self.length
+        x = 0
+        while x < n
+            yield(self[x], x)
+            x += 1
+        end
+         self
+    end
+    def my_select
+        return to_enum unless block_given?
+        pick = []
+        self.my_each { |x|
+            if yield(x)
+                pick.push(x)
+            end
+        }
+         pick
+    end
+    def my_all?
+        return to_enum unless block_given?
+        self.my_each { |x|
+        if yield(x) == false
+            return false
+        end
+        }
+         true
+    end
+    #my_any?
+    def my_any?
+        return to_enum unless block_given?
+        self.my_each { |x|
+        if yield(x) == true
+            return true
+        end
+        }
+         false
+    end
+    #my_none?
+    def my_none?
+        return to_enum unless block_given?
+        self.my_each { |x|
+        if yield(x) == true
+            return false
+        end
+        }
+         true
+    end
+    #my_count
+    def my_count
+        return to_enum unless block_given?
+        counter = 0
+        n= self.length
+        self.my_each { |x|
+        if yield(x) == true
+            counter += 1
+        else
+            return n
+        end
+        }
+         counter
+    end
```

Comment:
* [ ] should have working `my_count` identical and returns the same thing as ruby's [`count`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-count). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_count.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
```ruby
[1,2,3,4,5,6,7].my_count != [1,2,3,4,5,6,7].count

[1,2,3,4,2,5,6,7].my_count(2) != [1,2,3,4,2,5,6,7].count(2)
```